pkg/config/trust_engine.go

```diff
@@ -0,0 +1,31 @@
+package config
+
+import (
+	"fmt"
+
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/trustengine"
+)
+
+func ResolveAllTrustEngineReferences(config *StepConfig, params []StepParameters, trustEngineConfiguration trustengine.TrustEngineConfiguration) {
+	for _, param := range params {
+		if ref := param.GetReference("trustEngine"); ref != nil {
+			if config.Config[param.Name] == "" {
+				resolveTrustEngineReference(ref, config, &piperhttp.Client{}, param, trustEngineConfiguration)
+			}
+		}
+	}
+}
+
+// resolveTrustEngineReference retrieves a secret from Vault trust engine
+func resolveTrustEngineReference(ref *ResourceReference, config *StepConfig, client *piperhttp.Client, param StepParameters, trustEngineConfiguration trustengine.TrustEngineConfiguration) {
+	token, err := trustengine.GetTrustEngineSecret(ref.Name, client, trustEngineConfiguration)
```

Comment:
I would simplify the naming here.

```suggestion
	token, err := trustengine.GetToken(ref.Name, client, trustEngineConfiguration)
```